backends-common/postgres/src/test/java/org/apache/james/backends/postgres/PostgresTableManagerTest.java

```diff
@@ -0,0 +1,310 @@
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.backends.postgres;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+
+import java.util.List;
+import java.util.function.Function;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.james.backends.postgres.utils.PostgresExecutor;
+import org.jooq.impl.DSL;
+import org.jooq.impl.SQLDataType;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import io.r2dbc.postgresql.PostgresqlConnectionConfiguration;
+import io.r2dbc.postgresql.PostgresqlConnectionFactory;
+import io.r2dbc.postgresql.api.PostgresqlResult;
+import io.r2dbc.spi.Connection;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+@Testcontainers
+public class PostgresTableManagerTest {
+
+    @Container
+    private static final GenericContainer<?> pgContainer = PostgresFixture.PG_CONTAINER.get();
+
+    private PostgresqlConnectionFactory connectionFactory;
+
+    @BeforeEach
+    void beforeAll() {
+        connectionFactory = new PostgresqlConnectionFactory(PostgresqlConnectionConfiguration.builder()
+            .host(pgContainer.getHost())
+            .port(pgContainer.getMappedPort(PostgresFixture.PORT))
+            .username(PostgresFixture.Database.DB_USER)
+            .password(PostgresFixture.Database.DB_PASSWORD)
+            .database(PostgresFixture.Database.DB_NAME)
+            .schema(PostgresFixture.Database.SCHEMA)
+            .build());
+    }
+
+    @AfterEach
+    void afterEach() {
+        // clean data
+        Flux.usingWhen(connectionFactory.create(),
+                connection -> Mono.from(connection.createStatement("DROP SCHEMA " + PostgresFixture.Database.SCHEMA + " CASCADE").execute())
+                    .then(Mono.from(connection.createStatement("CREATE SCHEMA " + PostgresFixture.Database.SCHEMA).execute()))
+                    .flatMap(PostgresqlResult::getRowsUpdated),
+                Connection::close)
+            .collectList()
+            .block();
+    }
+
+    Function<PostgresModule, PostgresTableManager> tableManagerFactory = module -> new PostgresTableManager(new PostgresExecutor(connectionFactory.create()
+        .map(c -> c)), module);
+
+    @Test
+    void initializeTableShouldSuccessWhenModuleHasSingleTable() {
+        String tableName = "tableName1";
+
+        PostgresTable table = PostgresTable.name(tableName)
+            .creteTableStep(dslContext -> dslContext.createTable(tableName)
```

Comment:
Question: why not make sure to use instead `createTableIfNotExist` (as I see in the ticket) so we don't need to catch already exist exception? Same for indexes? 

But I'm ok with this though, better safe than sorry later :)